cmake/scripts/expand_instantiations.cc

```diff
@@ -166,12 +166,15 @@ read_whole_file(std::istream &in)
       whole_file += remove_comments(line);
       whole_file += '\n';
     }
-  // substitute tabs by spaces, multiple spaces by single ones
-  for (unsigned int i = 0; i < whole_file.size(); ++i)
-    if (whole_file[i] == '\t')
-      whole_file[i] = ' ';
-  while (whole_file.find("  ") != std::string::npos)
-    whole_file.replace(whole_file.find("  "), 2, " ");
+  // substitute tabs by spaces
+  std::replace(whole_file.begin(), whole_file.end(), '\t', ' ');
+  // substitute multiple spaces by single ones
+  size_t position = whole_file.find("  ");
+  while (position != std::string::npos)
+    {
+      whole_file.replace(position, 2, " ");
```

Comment:
Hmm.. I can't find the overload you are referring to and the compiler doesn't like replacing the `const char*` by a `char` neither.

Reply:
You could save a line if you wrote this as
```
  while ((position = whole_file.find (...)) != std::string::npos)
```
I think this is also easier to read because it more closely mimicks a `for` loop.